lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -53,55 +55,68 @@ class MailboxController extends BaseController {
       if (success is GetAllMailboxSuccess) {
         _buildTree(success.folderMailboxList);
-        _setDefaultSelected(success.defaultMailboxList);
+        _setUpMailboxDefault(success.defaultMailboxList);
       }
     });
   }
 
   @override
-  void onDone() {
-  }
+  void onDone() {}
 
   @override
-  void onError(error) {
-  }
+  void onError(error) {}
 
   void _buildTree(List<PresentationMailbox> folderMailboxList) async {
     folderMailboxTree.value = await _treeBuilder.generateMailboxTree(folderMailboxList);
   }
 
-  void _setDefaultSelected(List<PresentationMailbox> defaultMailboxList) {
+  void _setUpMailboxDefault(List<PresentationMailbox> defaultMailboxList) {
     try {
       final mailboxDefault = defaultMailboxList
           .firstWhere((presentationMailbox) => presentationMailbox.role == Role(MailboxConstants.ROLE_DEFAULT));
       selectedMailbox.value = mailboxDefault;
-      mailboxDashBoardController.mailboxCurrent.value = mailboxDefault;
+
+      _updateMailboxCurrentInMailboxDashBoard(mailboxDefault);
+
     } catch (e) {}
+
+    goToThreadView();
+  }
+
+  void _updateMailboxCurrentInMailboxDashBoard(PresentationMailbox newPresentationMailbox) {
+    if (Get.isRegistered<MailboxDashBoardController>()) {
+      final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
+      mailboxDashBoardController.mailboxCurrent.value = newPresentationMailbox;
+    }
   }
 
-  SelectMode getSelectMode(PresentationMailbox presentationMailbox) {
-    return presentationMailbox.id == selectedMailbox.value.id
+  SelectMode getSelectMode(PresentationMailbox presentationMailbox, PresentationMailbox selectedMailbox) {
+    return presentationMailbox.id == selectedMailbox.id
       ? SelectMode.ACTIVE
       : SelectMode.INACTIVE;
   }
 
   void selectMailbox(
       BuildContext context,
-      PresentationMailbox presentationMailboxSelected,
-      {
-        required GlobalKey<ScaffoldState> keyWidgetMailboxDashBoard
-      }
+      PresentationMailbox presentationMailboxSelected
   ) {
     selectedMailbox.value = presentationMailboxSelected;
-    mailboxDashBoardController.mailboxCurrent.value = presentationMailboxSelected;
+
+    _updateMailboxCurrentInMailboxDashBoard(presentationMailboxSelected);
 
     if (responsiveUtils.isMobile(context)) {
-      goToThread(keyWidgetMailboxDashBoard: keyWidgetMailboxDashBoard);
+      final mailboxDashBoardController = Get.find<MailboxDashBoardController>();
+      mailboxDashBoardController.scaffoldKey.currentState?.openEndDrawer();
     }
+
+    goToThreadView();
   }
 
-  void goToThread({required GlobalKey<ScaffoldState> keyWidgetMailboxDashBoard}) {
-    keyWidgetMailboxDashBoard.currentState?.openEndDrawer();
+  void goToThreadView() {
+    if (Get.isRegistered<ThreadController>()) {
+      final threadController = Get.find<ThreadController>();
+      threadController.updateMailbox(presentationMailbox: selectedMailbox.value, accountId: accountId);
```

Comment:
As we agreed, we will not modify other controller from outside, except from father controller?
In this case, Mailbox, Thread, Email are the child controller of MailboxDashboard.